sample/kotlin/src/main/kotlin/com/datadog/android/sample/data/db/LocalDataSource.kt

```diff
@@ -0,0 +1,64 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sample.data.db
+
+import android.content.Context
+import android.preference.PreferenceManager
+import com.datadog.android.sample.data.db.realm.RealmDataSource
+import com.datadog.android.sample.data.db.room.RoomDataSource
+import com.datadog.android.sample.data.db.sqldelight.SqlDelightDataSource
+import com.datadog.android.sample.data.db.sqlite.SQLiteDataSource
+import com.datadog.android.sample.data.model.Log
+import com.datadog.android.sample.datalist.DataSourceType
+import io.reactivex.rxjava3.core.SingleSource
+import java.lang.IllegalArgumentException
+
+class LocalDataSource(
+    val context: Context
+) {
+
+    private val dataSource: DataSource by lazy {
+        val choice = PreferenceManager.getDefaultSharedPreferences(context)
```

Comment:
Please do not use this PreferenceManager as it is deprecated. I already created the `Preferences` class which uses the AndroidX PreferenceManager. Can you please add this logic in there.

Reply:
Also here you are persisting the choice but from what I could see you are not persisting the choice for the ImageLoader.